object-manipulation-2/src/pick.js

```diff
@@ -0,0 +1,15 @@
+/* exported pick */
+function pick(source, keys) {
+  var newObj = {};
+  for (var property in source) {
+    for (var i = 0; i < keys.length; i++) {
+      if (property === keys[i]) {
+        newObj[property] = source[property];
+      }
+      if (newObj[property] === undefined) {
+        delete newObj[property];
+      }
+    }
+  }
```

Comment:
You can simplify this a bit. Try only looping through each element of the `keys` array, and if `source` at `keys` at `i` is not undefined, assign the value of `source` at `keys` at `i` to `newObj` at `keys` at `i`.